staging/src/k8s.io/apiserver/pkg/server/config_selfclient.go

```diff
@@ -101,8 +101,20 @@ func LoopbackHostPort(bindAddress string) (string, string, error) {
 
 	// Value is expected to be an IP or DNS name, not "0.0.0.0".
 	if host == "0.0.0.0" {
-		// compare MaybeDefaultWithSelfSignedCerts which adds "localhost" to the cert as alternateDNS
 		host = "localhost"
+		// Get ip of local interface, but fall back to "localhost".
+		// Note that "localhost" is resolved with the external nameserver first with Go's stdlib.
+		// So if localhost.<yoursearchdomain> resolves, we don't get a 127.0.0.1 as expected.
+		addrs, err := net.InterfaceAddrs()
+		if err == nil {
+			for _, address := range addrs {
+				// check the address type and if it is not a loopback the display it
+				if ipnet, ok := address.(*net.IPNet); ok && ipnet.IP.IsLoopback() {
```

Comment:
Should it be `ipnet.IP.IsLoopback()` or `!ipnet.IP.IsLoopback()` as the comment implies?